### PR TITLE
closes #23514: add firefox wallpapers

### LINKED ISSUE #23514: Add Firefox wallpapers
The Firefox wallpapers should be imported as drawables and included in the APK.

### PR DESCRIPTION
**Pull request by MatthewTighe**
image

image

image

image

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: .experimenter.json [hunk 1]
```diff
@@ -1,12 +1,20 @@
 {
-  "search-term-groups": {
-    "description": "A feature allowing the grouping of URLs around the search term that it came from.",
+  "nimbus-validation": {
```
Comment (Amejia481) [inline .experimenter.json:2]: maybe this could be commit in separate PR to just have what it's related to this PR :)

Comment (MatthewTighe) [reply .experimenter.json:2]: i legitimately don't know how this gets sneaky in, i'm not trying to change this file lmao. i've done a few rebases and it keeps showing up from some commit

File: app/src/main/java/org/mozilla/fenix/wallpapers/Wallpaper.kt [hunk 2]
```diff
@@ -51,6 +49,13 @@ enum class WallpaperOrigin {
     REMOTE,
 }
 
+val Wallpaper.drawableId: Int get() = when (name) {
+    "amethyst" -> R.drawable.amethyst
+    "cerulean" -> R.drawable.cerulean
+    "sunrise" -> R.drawable.sunrise
+    else -> -1
+}
+
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/wallpapers/Wallpaper.kt:58]: I think this will only apply to Firefox wallpapers, right? Having different types for different wallpapers could make this a bit simpler to handle related https://github.com/mozilla-mobile/fenix/pull/23505/commits/730a7f7e1db1349a0823c84a34375bec4412ffb3#r796792287

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/wallpapers/Wallpaper.kt:58]: This way we could have different properties that make sense for specific wallpaper types

Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/wallpapers/Wallpaper.kt:58]: Oh yeah, thanks for the reminder! I'll make some changes around that

File: .experimenter.json (hunk 1 above)
Comment (MatthewTighe) [reply .experimenter.json]: Removed those changes

File: app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt [hunk 3]
```diff
@@ -189,12 +167,15 @@ class WallpaperManager(
...
-            landscapePath = "",
-            isDark = false,
+            name = "default",
             themeCollection = WallpaperThemeCollection.NONE
         )
+        val localWallpapers = listOf(
+            defaultWallpaper,
+            Wallpaper("amethyst", themeCollection = WallpaperThemeCollection.FIREFOX),
+            Wallpaper("cerulean", themeCollection = WallpaperThemeCollection.FIREFOX),
+            Wallpaper("sunrise", themeCollection = WallpaperThemeCollection.FIREFOX),
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt:177]: 👍🏽

File: app/src/main/java/org/mozilla/fenix/wallpapers/Wallpaper.kt (hunk 2 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/wallpapers/Wallpaper.kt:58]: We could evaluate that in follow up :)

**Review by Amejia481 — APPROVED** [review]
:shipit:

**Comment by Amejia481** [comment]
Looks like we would need to update some tests 

```
e: /home/runner/work/fenix/fenix/app/src/test/java/org/mozilla/fenix/wallpapers/WallpaperManagerTest.kt: (30, 63): Type mismatch: inferred type is WallpaperStorage but WallpaperDownloader was expected
e: /home/runner/work/fenix/fenix/app/src/test/java/org/mozilla/fenix/wallpapers/WallpaperManagerTest.kt: (30, 85): Too many arguments for public constructor WallpaperManager(settings: Settings, downloader: WallpaperDownloader, crashReporter: CrashReporter) defined in org.mozilla.fenix.wallpapers.WallpaperManager

```

File: .experimenter.json [hunk 4]
```diff
@@ -1,4 +1,20 @@
 {
+  "default-browser-message": {
+    "description": "A small feature allowing experiments on the placement of a default browser message.",
+    "hasExposure": true,
+    "exposureDescription": "",
```
Comment (Amejia481) [inline .experimenter.json:5]: Let's avoid commit this :)
